utils/prof_utils.py

```diff
@@ -0,0 +1,523 @@
+        maxima: list
+            A list of floats corresponding to the bin location of any profile maxima
+        minima: list
+            A list of floats correspodning to the bin location of any profile minima
+    """
+
+    #regularize the profile
+    profile = np.array(profile)/max(profile) #for meaningful noise mean and std prints
+    on_pulse_prof, noise_mean, noise_std = regularize_prof(profile, reg_param=reg_param)
+    logger.debug("Profile noise mean: {}".format(noise_mean))
+    logger.debug("Profile noise STD: {}".format(noise_std))
+
+    #find widths:
+    W10, W50, Weq = find_widths(on_pulse_prof)
+    #Width uncertainties: Find the average difference between widths the profiles with noise std added and subtracted. Then add 0.5 bins
+    less_std_prof, _, _  = regularize_prof(np.array(profile) - noise_std, reg_param=reg_param)
+    W10_less, W50_less, Weq_less = find_widths(less_std_prof)
+    more_std_prof, _, _ = regularize_prof(np.array(profile) + noise_std, reg_param=reg_param)
+    W10_more, W50_more, Weq_more = find_widths(more_std_prof)
+    W10_e = (abs(W10-W10_less) + abs(W10-W10_more))/2 + 0.5
+    W50_e = (abs(W50-W50_less) + abs(W50-W50_more))/2 + 0.5
+    Weq_e = (abs(Weq-Weq_less) + abs(Weq-Weq_more))/2 + 0.5
+
+    #find max, min. Root uncertainties are 0.5 bins
+    minima, maxima = find_minima_maxima(on_pulse_prof, ignore_threshold=ignore_threshold, min_comp_len=min_comp_len)
+
+    logger.info("W10:                   {0} +/- {1}".format(W10, W10_e))
+    logger.info("W50:                   {0} +/- {1}".format(W50, W50_e))
+    logger.info("Weq:                   {0} +/- {1}".format(Weq, Weq_e))
+    logger.info("Maxima:                {0}".format(maxima))
+    logger.info("Minima:                {0}".format(minima))
+
+    #plotting
+    if plot_name:
+        std_prof = np.array(profile) - noise_mean
+        std_prof = std_prof/np.max(std_prof)
+        plt.figure(figsize=(20, 12))
+        plt.plot(on_pulse_prof, label="Regularized profile")
+        plt.plot(std_prof, label="Original Profile")
+        plt.title(plot_name.split("/")[-1], fontsize=22)
+        plt.xticks(fontsize=18)
+        plt.yticks(fontsize=18)
+        plt.xlim(0, len(std_prof))
+        plt.xlabel("Bins", fontsize=20)
+        plt.ylabel("Intensity", fontsize=20)
+        for mx in maxima:
+            plt.axvline(x=mx, ls=":", lw=2, color="gray")
+        plt.legend(loc="upper right", prop={'size': 16})
+        plt.savefig(plot_name)
+
+    return [W10, W10_e, W50, W50_e, Weq, Weq_e, maxima, minima]
+
+if __name__ == '__main__':
+
+    loglevels = dict(DEBUG=logging.DEBUG,\
+                    INFO=logging.INFO,\
+                    WARNING=logging.WARNING,\
+                    ERROR=logging.ERROR)
+
+    parser = argparse.ArgumentParser(description="""A utility file for calculating a variety of pulss profile properties""", formatter_class=argparse.ArgumentDefaultsHelpFormatter)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [line too long (180 > 159 characters)](https://app.codacy.com/app/CIRA-Pulsars-and-Transients-Group/vcstools/pullRequest?prid=4645369)